alf/environments/suite_metadrive.py

```diff
@@ -74,10 +94,19 @@ def __init__(self,
         self._image_channel_first = image_channel_first
         self._observation_spec = _space_to_spec(self._env.observation_space)
 
+        self._vel_encoding = vel_encoding
+        if self._vel_encoding is not None:
+            h, w, c = self._observation_spec.shape
+            self._observation_spec = BoundedTensorSpec(
```

Comment:
An alternative approach could be to fuse at the mode side (e.g. by expanding the scalar velocity to an image and concat), which has the advantage of using less memory.